### PR TITLE
Fixes CC export when students have nil names

### LINKED ISSUE #1093: Export fails when a student has a nil last_name
Should probably disallow nil first_name, last_name, full_name (remove?), title.
Use "" instead of nil.

### PR DESCRIPTION
**Pull request by Dantemss**
Also moved the spec to a folder matching the class structure

Closes #1093

### CODE REVIEW

File: app/subsystems/tasks/get_tp_performance_report.rb [hunk 1]
```diff
@@ -23,41 +23,39 @@ def exec(course:)
...
+          {
+            name: student_role.name,
+            first_name: student_role.first_name,
+            last_name: student_role.last_name,
+            student_identifier: student_role.student.student_identifier,
+            role: student_role.id,
+            data: data,
+            average_score: average_scores(data.map{|datum| datum.present? ? datum[:task] : nil})
+          }
+        end.compact
```
Comment (Dantemss) [inline app/subsystems/tasks/get_tp_performance_report.rb]: Sorry, this is all just indentation.

**Comment by Dantemss** [comment]
Sorting is already done in https://github.com/openstax/tutor-server/blob/master/app/subsystems/tasks/get_cc_performance_report.rb#L37

**Comment by jpslav** [comment]
I might just put into my export PR

**Comment by jpslav** [comment]
Replaced by https://github.com/openstax/tutor-server/pull/1118 (master already had files moved, etc, and was easier to make a new one than rebase).  Thanks @Dantemss for the original.

**Pull request closed without merging**